Mathlib/Algebra/GroupWithZero/NonZeroDivisors.lean

```diff
@@ -277,3 +275,78 @@ lemma nonZeroSMulDivisors_mulOpposite_eq_op_nonZeroDivisors :
     Rᵐᵒᵖ ⁰[R] = R⁰.op := rfl
 
 end nonZeroSMulDivisors
+
+noncomputable section Associated
+
+open nonZeroDivisors
+
+/-- The units of the monoid of non zero divisors of `α` are equivalent to the units of `α`. -/
+def nonZeroDivisorsUnitsEquiv (α : Type*) [MonoidWithZero α] :
+    (α⁰)ˣ ≃* αˣ :=
+  MulEquiv.ofBijective (Units.map α⁰.subtype) ⟨Units.map_injective Subtype.val_injective,
+    fun u ↦ ⟨IsUnit.unit ⟨⟨⟨u, IsUnit.mem_nonZeroDivisors u.isUnit⟩, ⟨(u⁻¹ : αˣ),
+        IsUnit.mem_nonZeroDivisors u⁻¹.isUnit⟩, by simp, by simp⟩, rfl⟩,
+      by rw [Units.ext_iff, IsUnit.unit_of_val_units, Units.coe_map, Submonoid.coe_subtype]⟩⟩
+
+@[simp]
+theorem nonZeroDivisorsUnitsEquiv_apply (α : Type*) [MonoidWithZero α] (u : (α⁰)ˣ) :
+    nonZeroDivisorsUnitsEquiv α u = (u : α) := rfl
+
+@[simp]
+theorem nonZeroDivisorsUnitsEquiv_symm_apply (α : Type*) [MonoidWithZero α] (u : αˣ) :
+    ((nonZeroDivisorsUnitsEquiv α).symm u : α) = (u : α) := by
+  obtain ⟨v, rfl⟩ := (nonZeroDivisorsUnitsEquiv α).surjective u
+  rw [MulEquiv.symm_apply_apply, nonZeroDivisorsUnitsEquiv_apply]
+
+theorem Associates_mk_mem_nonZeroDivisors_iff {α : Type*} [CommMonoidWithZero α] (a : α) :
```

Comment:
```suggestion
theorem associatesMk_mem_nonZeroDivisors_iff {α : Type*} [CommMonoidWithZero α] (a : α) :
```
or just
```suggestion
theorem mk_mem_nonZeroDivisors_iff {α : Type*} [CommMonoidWithZero α] (a : α) :
```
since which `mk` it is is clear from context